### DISCUSSION TITLE
initialisation depencencies issues

### DISCUSSION THREAD

**Discussion by stephane303** [vuejs/pinia#550] [Help and Questions]
I am having trouble dealing with initialisation dependencies:

For example:

I have two stores:
a) settingsStore 
b) menuStore

settingsStore need to read some configuration from the DB.
menuStore depends on the states of settingsStore (received from the DB).

How can I make menuStore wait for settingsStore to have finished initialisation ?

**Comment by wujekbogdan** [comment]
I would use dependency injection - wrap stores in a function that initializes the `settingsStore`, waits for an action that initializes the state to resolve and then pass that store to `menuStore` via argument.

I do something pretty similar in my app - see the code sample here: https://github.com/posva/pinia/issues/557
